spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java

```diff
@@ -307,6 +341,17 @@ private FetchResult fetch(Git git, String label) {
 			}
 			return result;
 		}
+		catch (TransportException ex){
+			if(errorLogOnFailingFetch){
```

Comment:
I don't think I like a branch here. This is too into the weeds for a configuration item. If you're going to log something, pick the one that works and be done.